drivers/nutdrv_qx.c

```diff
@@ -859,14 +910,59 @@ static int	fabula_command(const char *cmd, char *buf, size_t buflen)
 
 	upsdebugx(4, "command index: 0x%02x", index);
 
+	if (hunnox_patch) {
+		// Enable lock-step protocol for Hunnox
+		if  (hunnox_protocol(index) != 0) {
+			return 0;
+		}
+
+		// Seems that if we inform a large buffer, the USB locks.
+		// This value was captured from the Windows "official" client.
+		if (buflen > 102) {
+			buflen = 102;
+		}
+	}
+
 	/* Send command/Read reply */
-	ret = usb_get_string_simple(udev, index, buf, buflen);
+	if (langid_fix != -1) {
```

Comment:
It is important to determine whether additional conditions here do not affect other sub-drivers or this protocol.

Reply:
Constrained to `&& hunnox_patch`

Reply:
UPDATE: Ended up separating original `fabula_command()` from `fabula_hunnox_command()` and dropping the `hunnox_patch` value support, so restoring the originally PRed clause for this choice point.